docs/changelog/4_x.rst

```diff
@@ -0,0 +1,467 @@
+"sha512". Moreover, any digest algorithm OpenSSL supports is automatically
+picked up.
+
+
+pg_trgm
+-------
+
+This extension provides a bunch of functionality used to determine string
+similarity, which makes it a good text search alternative for some use cases:
+
+edgedb> with x := {'hello world', 'word hero', 'help the world'}
+....... select res := (x, ext::pg_trgm::word_similarity(x, 'hello world'))
+....... order by res.1 desc;
+{('hello world', 1), ('help the world', 0.5), ('word hero', 0.35714287)}
+
+
+
+Additional changes
+==================
+
+EdgeQL
+------
+
+* Add ``anyobject`` pseudo type.
+  (:eql:gh:`#5961`)
+
+  It is a placeholder mainly used in polymorphic functions. It denotes any
+  object type and like :eql:type:`anytype` it preserves the actual type if the
+  placeholder ``anyobject`` appears in the function return signature.
+
+* Add new style of ``if``/``then``/``else`` syntax.
+  (:eql:gh:`#6074`)
+
+  Many people find it more natural to write "if ... then .. else ..." for
+  conditional expressions because it mirrors the conditional statement from
+  other familiar programming languages.
+
+  .. code-block:: edgeql-repl
+
+    edgedb> select if count(Object) > 0 then 'got data' else 'no data';
+    {'got data'}
+
+* Support conditional DML.
+  (:eql:gh:`#4437`)
+
+  It can be useful to be able to create, update or delete different objects
+  based on some condition:
+
+  .. code-block:: edgeql
+
+    with
+      name := <str>$0,
+      admin := <bool>$1
+    select if admin then (
+        insert AdminUser { name := name }
+    ) else (
+        insert User { name := name }
+    )
+
+  A different use-case of conditional DML is using a :eql:op:`coalesce`
+  operator to express things like "select or insert if missing":
+
+  .. code-block:: edgeql
+
+    select (select User filter .name = 'Alice') ??
+           (insert User { name := 'Alice' });
+
+* Add ``contains`` for JSON so that it can be used with ``pg::gin`` index.
+  (:eql:gh:`#5703`)
+
+* Add :eql:func:`to_bytes` to convert :eql:type:`str` into :eql:type:`bytes`
+  using UTF-8 encoding.
+  (:eql:gh:`#4281`)
+
+* Add :eql:func:`to_str` to convert :eql:type:`bytes` into :eql:type:`str`
+  using UTF-8 encoding.
+  (:eql:gh:`#4281`)
+
+* Add ``enc::base64_encode`` and ``enc::base64_decode`` functions.
+  (:eql:gh:`#5963`)
+
+  .. code-block:: edgeql-repl
+
+    edgedb> select enc::base64_encode(b'hello');
+    {'aGVsbG8='}
+    edgedb> select enc::base64_decode('aGVsbG8=');
+    {b'hello'}
+
+* Support optional config values.
+  (:eql:gh:`#6025`)
+
+* Add ``when`` clause to triggers to enable them to be conditional.
+  (:eql:gh:`#5994`)
+
+* Allow empty arrays without cast in ``insert``.
+  (:eql:gh:`#6218`)
+
+
+
+Bug fixes
+---------
+
+* Fix a casting bug for some aliased expressions.
+  (:eql:gh:`#5788`)
+
+* Fix cardinality inference of calls to functions with ``optional`` args.
+  (:eql:gh:`#5867`)
+
+* Fix duplicating outputs for some tuples.
```

Comment:
I'd drop this one, it's a generated code quality tweak that shouldn't impact query perf